Rename hasTransform() to isTransformed() on RenderObject and RenderLayer

graouts opened the pull request:
#### c934d4911662a8068e232e316860641e7ccbe020
<pre>
Rename hasTransform() to isTransformed() on RenderObject and RenderLayer
<a href="https://bugs.webkit.org/show_bug.cgi?id=250909">https://bugs.webkit.org/show_bug.cgi?id=250909</a>

Reviewed by Simon Fraser.

The name &quot;hasTransform&quot; is kind of convoluted in WebCore due to historically denoting on
RenderStyle whether a &quot;transform&quot; property was set to some non-none value. With the
addition of more CSS properties affecting transform, and the addition of
RenderStyle::affectsTransform(), we should also rename &quot;hasTransform&quot; on
RenderLayer and RenderObject to be &quot;isTransformed&quot; to better match what this
method denotes.

* Source/WebCore/display/css/DisplayBox.h:
* Source/WebCore/rendering/BackgroundPainter.cpp:
(WebCore::BackgroundPainter::calculateBackgroundImageGeometry):
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::simplifiedLayout):
(WebCore::RenderBlock::isSelectionRoot const):
(WebCore::RenderBlock::selectionGaps):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::layoutBlock):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::isCandidateForOpaquenessTest):
(WebCore::RenderBox::layoutOverflowRectForPropagation const):
* Source/WebCore/rendering/RenderDeprecatedFlexibleBox.cpp:
(WebCore::RenderDeprecatedFlexibleBox::layoutBlock):
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
(WebCore::RenderFlexibleBox::layoutBlock):
* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(WebCore::RenderFragmentContainer::layoutOverflowRectForBoxForPropagation):
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::layoutBlock):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::layoutShadowContent):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::updateTransform):
(WebCore::RenderLayer::updatePagination):
(WebCore::RenderLayer::cannotBlitToWindow const):
(WebCore::transparencyClipBox):
(WebCore::RenderLayer::collectFragments):
* Source/WebCore/rendering/RenderLayer.h:
(WebCore::RenderLayer::canUseOffsetFromAncestor const):
(WebCore::RenderLayer::isTransformed const):
(WebCore::RenderLayer::hasTransform const): Deleted.
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateTransform):
(WebCore::layerOrAncestorIsTransformedOrUsingCompositedScrolling):
(WebCore::RenderLayerBacking::getCurrentTransform const):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::reasonsForCompositing const):
(WebCore::RenderLayerCompositor::requiresCompositingForTransform const):
* Source/WebCore/rendering/RenderLayerModelObject.cpp:
(WebCore::RenderLayerModelObject::styleWillChange):
(WebCore::RenderLayerModelObject::styleDidChange):
(WebCore::RenderLayerModelObject::mapLocalToSVGContainer const):
* Source/WebCore/rendering/RenderLayerModelObject.h:
* Source/WebCore/rendering/RenderLayoutState.cpp:
(WebCore::shouldDisablePaintOffsetCacheForSubtree):
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::shouldUseTransformFromContainer const):
(WebCore::RenderObject::pushOntoGeometryMap const):
(WebCore::RenderObject::offsetFromAncestorContainer const):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isTransformed const):
(WebCore::RenderObject::hasTransformOrPerspective const):
(WebCore::RenderObject::hasTransform const): Deleted.
* Source/WebCore/rendering/RenderTable.cpp:
(WebCore::RenderTable::layout):
* Source/WebCore/rendering/RenderTableRow.cpp:
(WebCore::RenderTableRow::layout):
* Source/WebCore/rendering/RenderTableSection.cpp:
(WebCore::RenderTableSection::calcRowLogicalHeight):
(WebCore::RenderTableSection::layout):
(WebCore::RenderTableSection::layoutRows):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::documentRect const):
* Source/WebCore/rendering/svg/RenderSVGModelObject.cpp:
(WebCore::RenderSVGModelObject::mapAbsoluteToLocalPoint const):
* Source/WebCore/rendering/svg/SVGBoundingBoxComputation.cpp:
(WebCore::SVGBoundingBoxComputation::handleRootOrContainer const):
* Source/WebKitLegacy/mac/DOM/DOMUIKitExtensions.mm:
(-[DOMHTMLElement structuralComplexityContribution]):

Canonical link: <a href="https://commits.webkit.org/259171@main">https://commits.webkit.org/259171@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/e31ccefa5d4bf3f00f613cb04a4b0b400f0945f2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/104094 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/13186 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/37018 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/113303 "Built successfully") | [❌ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/173611 "Failed to checkout and rebase branch from PR 8886") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/108036 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/14252 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/4101 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/96325 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/112377 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/109864 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/10981 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/94042 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/38653 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/92832 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/25654 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/80309 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/6554 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/27030 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/6696 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/3570 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/12696 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/46566 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/8474 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3349 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

alanbaradlay commented on Source/WebCore/rendering/RenderObject.h:
```diff
@@ -479,8 +479,8 @@ class RenderObject : public CachedImageClient {
     bool hasTransformRelatedProperty() const { return m_bitfields.hasTransformRelatedProperty(); } // Transform, perspective or transform-style: preserve-3d.
-    bool hasTransform() const { return hasTransformRelatedProperty() && (style().hasTransform() || style().translate() || style().scale() || style().rotate() || hasSVGTransform()); }
-    bool hasTransformOrPerspective() const { return hasTransformRelatedProperty() && (hasTransform() || style().hasPerspective()); }
+    bool isTransformed() const { return hasTransformRelatedProperty() && (style().affectsTransform() || hasSVGTransform()); }
```
I am not sure if isTransformed() is any better (like what does it mean if a renderer is transformed) but I don't have a better idea either so..

graouts committed:
Rename hasTransform() to isTransformed() on RenderObject and RenderLayer
https://bugs.webkit.org/show_bug.cgi?id=250909

Reviewed by Simon Fraser.

The name "hasTransform" is kind of convoluted in WebCore due to historically denoting on
RenderStyle whether a "transform" property was set to some non-none value. With the
addition of more CSS properties affecting transform, and the addition of
RenderStyle::affectsTransform(), we should also rename "hasTransform" on
RenderLayer and RenderObject to be "isTransformed" to better match what this
method denotes.

* Source/WebCore/display/css/DisplayBox.h:
* Source/WebCore/rendering/BackgroundPainter.cpp:
(WebCore::BackgroundPainter::calculateBackgroundImageGeometry):
* Source/WebCore/rendering/RenderBlock.cpp:
(WebCore::RenderBlock::simplifiedLayout):
(WebCore::RenderBlock::isSelectionRoot const):
(WebCore::RenderBlock::selectionGaps):
* Source/WebCore/rendering/RenderBlockFlow.cpp:
(WebCore::RenderBlockFlow::layoutBlock):
* Source/WebCore/rendering/RenderBox.cpp:
(WebCore::isCandidateForOpaquenessTest):
(WebCore::RenderBox::layoutOverflowRectForPropagation const):
* Source/WebCore/rendering/RenderDeprecatedFlexibleBox.cpp:
(WebCore::RenderDeprecatedFlexibleBox::layoutBlock):
* Source/WebCore/rendering/RenderFlexibleBox.cpp:
(WebCore::RenderFlexibleBox::layoutBlock):
* Source/WebCore/rendering/RenderFragmentContainer.cpp:
(WebCore::RenderFragmentContainer::layoutOverflowRectForBoxForPropagation):
* Source/WebCore/rendering/RenderGrid.cpp:
(WebCore::RenderGrid::layoutBlock):
* Source/WebCore/rendering/RenderImage.cpp:
(WebCore::RenderImage::layoutShadowContent):
* Source/WebCore/rendering/RenderLayer.cpp:
(WebCore::RenderLayer::updateTransform):
(WebCore::RenderLayer::updatePagination):
(WebCore::RenderLayer::cannotBlitToWindow const):
(WebCore::transparencyClipBox):
(WebCore::RenderLayer::collectFragments):
* Source/WebCore/rendering/RenderLayer.h:
(WebCore::RenderLayer::canUseOffsetFromAncestor const):
(WebCore::RenderLayer::isTransformed const):
(WebCore::RenderLayer::hasTransform const): Deleted.
* Source/WebCore/rendering/RenderLayerBacking.cpp:
(WebCore::RenderLayerBacking::updateTransform):
(WebCore::layerOrAncestorIsTransformedOrUsingCompositedScrolling):
(WebCore::RenderLayerBacking::getCurrentTransform const):
* Source/WebCore/rendering/RenderLayerCompositor.cpp:
(WebCore::RenderLayerCompositor::reasonsForCompositing const):
(WebCore::RenderLayerCompositor::requiresCompositingForTransform const):
* Source/WebCore/rendering/RenderLayerModelObject.cpp:
(WebCore::RenderLayerModelObject::styleWillChange):
(WebCore::RenderLayerModelObject::styleDidChange):
(WebCore::RenderLayerModelObject::mapLocalToSVGContainer const):
* Source/WebCore/rendering/RenderLayerModelObject.h:
* Source/WebCore/rendering/RenderLayoutState.cpp:
(WebCore::shouldDisablePaintOffsetCacheForSubtree):
* Source/WebCore/rendering/RenderObject.cpp:
(WebCore::RenderObject::shouldUseTransformFromContainer const):
(WebCore::RenderObject::pushOntoGeometryMap const):
(WebCore::RenderObject::offsetFromAncestorContainer const):
* Source/WebCore/rendering/RenderObject.h:
(WebCore::RenderObject::isTransformed const):
(WebCore::RenderObject::hasTransformOrPerspective const):
(WebCore::RenderObject::hasTransform const): Deleted.
* Source/WebCore/rendering/RenderTable.cpp:
(WebCore::RenderTable::layout):
* Source/WebCore/rendering/RenderTableRow.cpp:
(WebCore::RenderTableRow::layout):
* Source/WebCore/rendering/RenderTableSection.cpp:
(WebCore::RenderTableSection::calcRowLogicalHeight):
(WebCore::RenderTableSection::layout):
(WebCore::RenderTableSection::layoutRows):
* Source/WebCore/rendering/RenderView.cpp:
(WebCore::RenderView::documentRect const):
* Source/WebCore/rendering/svg/RenderSVGModelObject.cpp:
(WebCore::RenderSVGModelObject::mapAbsoluteToLocalPoint const):
* Source/WebCore/rendering/svg/SVGBoundingBoxComputation.cpp:
(WebCore::SVGBoundingBoxComputation::handleRootOrContainer const):
* Source/WebKitLegacy/mac/DOM/DOMUIKitExtensions.mm:
(-[DOMHTMLElement structuralComplexityContribution]):

Canonical link: https://commits.webkit.org/259171@main
